recipes/libpq/all/conanfile.py

```diff
@@ -56,10 +56,11 @@ def configure(self):
         if self.settings.compiler != "Visual Studio" and self.settings.os == "Windows":
             if self.options.shared:
                 raise ConanInvalidConfiguration("static mingw build is not possible")
+        # Looking into source code, it appears that zlib is not used in libpq
+        if self.options.with_zlib != "deprecated":
+            self.output.warn("with_zlib option is deprecated, do not use anymore")
 
```

Comment:
By removing the `with_zlib` option, `libpq` packages configured with different option values, will be aliased to the same package id. Which is correct as they are the same.

```suggestion
            self.output.warn("with_zlib option is deprecated, do not use anymore")
        del self.options.with_zlib
```